src/logic.service.ts

```diff
@@ -188,6 +187,77 @@ export const handleGetConditionsForEntity = async (params: GetEntityConditions):
   }
 };
 
+export const handleUpdateExpiryDateForConditionsOfEntity = async (
+  params: ConditionRequest,
+  xprtnDtTm: string,
+): Promise<{ code: number; message: string }> => {
+  if (!xprtnDtTm) {
+    xprtnDtTm = new Date().toISOString();
+  } else {
+    if (hasDateExpired(new Date(xprtnDtTm))) {
+      return { code: 400, message: 'Expiration time date provided was before the current time date.' };
+    }
+
+    if (!isDateValid(xprtnDtTm)) {
+      return { code: 400, message: 'Expiration time date provided was invalid.' };
+    }
+  }
```

Comment:
There is no need to check a non optional parameter as this the only scenario where xprtnDtTm will fail if it is will not parse as a date or is set to `''` which you can check for both in the `isDateValid` function. (Which should happen first)

```suggestion
  if(!isDateValid(xprtnDtTm)) {...} // covers what I highlighted above. Should also happen first
  const date = new Date(xprtnDtTm);
  if(hasDateExpired(date)) {...}// expired logic
  // normal execution  
```